### PR TITLE
Add workflow for checking document links

### PR DESCRIPTION
**Pull request by hui1110**
As title.
Includes the following parts:
- Regularly check the current [Spring Cloud Azure Reference Documentation](https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html) for broken links.
- Trigger workflow on push event to check generated HTML file for broken links, but only for main branch.

### CODE REVIEW

File: .github/workflows/brokenLinkCheck.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Broken Link Checker
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
```
Comment (saragluna) [inline .github/workflows/brokenLinkCheck.yml:8]: should there be a `pull_request` trigger too?

Comment (saragluna) [reply .github/workflows/brokenLinkCheck.yml:8]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

File: .github/workflows/scheduleCurrentLinkCheck.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  link-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: technote-space/broken-link-checker-action@v1.1.3
+        with:
+          TARGET: https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html
```
Comment (saragluna) [inline .github/workflows/scheduleCurrentLinkCheck.yaml:17]: Will it work if we change to `https://microsoft.github.io/spring-cloud-azure/`?

File: .github/workflows/scheduleCurrentLinkCheck.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Schedule Current Link Check
+
+# Controls when the workflow will run
+on:
+  # The schedule event allows you to trigger a workflow at a scheduled time.
+  schedule:
```
Comment (saragluna) [inline .github/workflows/scheduleCurrentLinkCheck.yaml:8]: let's add one https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch trigger too

**Review by saragluna — CHANGES_REQUESTED** [review]

File: .github/workflows/brokenLinkCheck.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          mvn clean install
+      - name: Link Checker
+        id: link-report
+        uses: peter-evans/link-checker@v1
+        with:
+          args: -d --document-root . -r -x "localhost|127.0.0.1"
+      - name: Get the result
+        run: |
+          errors=`cat link-checker/out.md | grep 'ERROR' | wc -l`
```
Comment (chenrujun) [inline .github/workflows/brokenLinkCheck.yml:31]: How about print these errors in the GitHub Actions log?

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: .github/workflows/brokenLinkCheck.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  check_html_links_job:
+    runs-on: ubuntu-latest
+    name: check-html-links-action
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build Docs
+        run: |
+          mvn clean install
```
[the remark this answers is not in the document]
Comment (hui1110) [reply .github/workflows/brokenLinkCheck.yml:23]: updated

**Comment by chenrujun** [comment]
To fix the pipeline failure, please add this step: https://github.com/microsoft/spring-cloud-azure/blob/44681d5126919a21e20638b7b927ceb9be521cef/.github/workflows/updateDocs.yaml#L24-L41

File: .github/workflows/brokenLinkCheck.yml (hunk 1 above)
Comment (chenrujun) [reply .github/workflows/brokenLinkCheck.yml:8]: @hui1110 ping.

File: .github/workflows/scheduleCurrentLinkCheck.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  link-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: technote-space/broken-link-checker-action@v1.1.3
+        with:
+          TARGET: https://microsoft.github.io/spring-cloud-azure/
+          EXCLUDED_KEYWORDS: localhost,127.0.0.1
+          EXCLUDE_LINKS_TO_SAME_PAGE: false
```
[the remark this answers is not in the document]
Comment (chenrujun) [reply .github/workflows/scheduleCurrentLinkCheck.yaml:20]: @hui1110 , ping.

**Comment by chenrujun** [comment]
@hui1110 , 
1. https://github.com/microsoft/spring-cloud-azure/runs/5882788678?check_suite_focus=true#step:6:7 now there are a lot of errors int the GitHub Action log, but it still passed, could you please investigate this?
2. How about block the PR if the `Check links Action` is failed?
image

**Pull request merged**